parsing/builtin_attributes.ml

```diff
@@ -92,15 +92,17 @@ let builtin_attrs =
 
 let is_builtin_attr s = Hashtbl.mem builtin_attrs s
 
+let track = ref true
+let track_attrs b = track := b
```

Comment:
What about

```ocaml
val track_attrs_in : track_attrs:bool -> (unit -> 'a) -> 'a
```

that uses `protect_refs`? Seems considerably safer and easier to reason about then the current very stateful approach. I'm not really objecting to the state, per se, but just the interface to interacting with it.